R/createQuadrantGate.R

```diff
@@ -0,0 +1,95 @@
+#' Create quadrant gate
+#'
+#' Creates a quadrant gate. Quadrant gates have four sectors (upper-right,
+#' upper-left, lower-left, lower-right), each with a unique gid and name.
+#'
+#' @param experimentId The ID of the experiment to which to add the gate.
+#' @param xChannel The name of the x channel to which the gate applies.
+#' @param yChannel The name of the y channel to which the gate applies.
+#' @param name The name of the gate. Each sector is named with this parameter
+#' and a quadrant flag (UR, UL, LL, LR).
+#' @param x The x coordinate (after the channel's scale has been applied).
+#' @param y The y coordinate (after the channel's scale has been applied).
+#' @param angles Angles at which the quadrant lines appear, in radians.
+#' Zero (0) points horizontally to the right; angles proceed counter-clockwise.
+#' Currently these must be 0, pi / 2, pi and 3 * pi / 2.
+#' @param gid Top-level group ID of the gate, used for tailoring. If this is not specified,
+#'   then a new Group ID will be created. For compound gates, "gid" refers to the
+#'   top-level GID. Each sector has a unique `model.gid` and `name` to which
+#'   populations must refer.
+#' @param gids Group IDs of each sector, assigned to `model.gid`.
+#' @param parentPopulationId ID of the parent population. Use \code{NULL} for
+#'   the "ungated" population. If specified, do not specify \code{parentPopulation}.
+#' @param parentPopulation Name of the parent population. An attempt will be made
+#'   to find the population by name. If zero or more than one population exists
+#'   with the name, an error will be thrown. If specified, do not specify
+#'   \code{parentPopulationId}.
+#' @param tailoredPerFile Whether this gate is tailored per FCS file.
+#' @param fcsFileId ID of FCS file, if tailored per file. Use \code{NULL} for
+#'   the global gate in a tailored gate group. If specified, do not specify
+#'   \code{fcsFile}.
+#' @param fcsFile Name of FCS file, if tailored per file. An attempt will be made
+#'   to find the file by name. If zero or more than one file exists with the name,
+#'   an error will be thrown. If specified, do not specify \code{fcsFileId}.
+#' @param locked Prevents modification of the gate via the web interface.
+#' @param createPopulation Automatically create corresponding population.
+#' @export
+#' @examples
+#' \dontrun{
+#' createQuadrantGate(experimentId, "FSC-A", "FSC-W", "my gate", 160000, 200000)
+#' }
+createQuadrantGate = function(experimentId, xChannel, yChannel, name,
+                               x, y, angles = c(0.000000, 3/2*pi, pi, pi/2),
```

Comment:
Our UI codes goes CCW -- can you reverse the angles please?

For now you can also remove this as a parameter and hard-code it into the function body.